library/old_protocol/old_protocol.cpp

```diff
@@ -0,0 +1,204 @@
+#include "old_protocol.hpp"
+
+#include <blackhole/attribute.hpp>
+
+#include "deserialize.hpp"
+#include "library/common.hpp"
+#include "library/elliptics.h"
+#include "library/logger.hpp"
+
+namespace ioremap { namespace elliptics { namespace n2 {
+
+old_protocol::old_protocol()
+: lookup_request_translator_(subscribe_table_[DNET_CMD_LOOKUP_NEW])
+{}
+
+void old_protocol::subscribe_request(int cmd, on_request_t on_request) {
+	subscribe_table_.at(cmd) = std::move(on_request);
+}
+
+void old_protocol::send_request(dnet_net_state *st,
+                                std::unique_ptr<n2_request> request,
+                                n2_repliers repliers) {
+	auto &cmd = request->cmd;
+	repliers_table_.emplace(uint64_t(cmd.trans), std::move(repliers));
+
+	switch (cmd.cmd) {
+	case DNET_CMD_LOOKUP_NEW:
+		enqueue_net(st, serialize_lookup_request(std::move(request)));
+		break;
+	default:
+		throw std::domain_error("old_protocol::send_request is not supported for "
+		                        "cmd = " + std::to_string(cmd.cmd));
+	}
+}
+
+std::unordered_set<int> old_protocol::supported_requests = { DNET_CMD_LOOKUP_NEW };
+std::unordered_set<int> old_protocol::supported_replies_on_srv = { DNET_CMD_LOOKUP_NEW };
+std::unordered_set<int> old_protocol::supported_replies_on_cl = {};
+
+int old_protocol::recv_message(dnet_cmd *cmd, dnet_net_state *st) {
+	if (!is_supported_message(cmd, st))
```

Comment:
Let's simplify this condition - creating lambda and unordered_set each time is too heavy.